rts/motoko-rts/src/region.rs

```diff
@@ -948,6 +948,19 @@ pub(crate) unsafe fn region_load<M: Memory>(_mem: &mut M, r: Value, offset: u64,
     }
 }
 
+pub unsafe fn region_write_heap_shapshot<M: Memory>(mem: &mut M, r: Value) {
+    use crate::constants::WASM_PAGE_SIZE;
+    use core::arch::wasm32;
+
+    let heap_size: usize = wasm32::memory_size(0) * (WASM_PAGE_SIZE.as_u32() as usize);
+    if wasm32::memory_size(0) as u64 > region_size(mem, r) {
+        let new_pages = wasm32::memory_size(0) as u64 - region_size(mem, r);
+        region_grow(mem, r, new_pages);
+    };
+    let heap_src = core::slice::from_raw_parts(core::ptr::null(), heap_size);
```

Comment:
I found that slices can't be big enough to copy up to 4GB of data (that's why Blob's are read/written in two halves). I think the size argument is signed, so at most 2^31-1 (Int32::MAX). 

Not sure, but eventually, we may have to hard-code this function in Wasm to avoid having a rust frame for the function on the Rust shadow stack.